assets/js/util/report-validation.test.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * Tests for reporting API validation utilities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	isValidDateRange,
+	isValidDimensions,
+	isValidMetrics,
+	isValidOrders,
+} from './report-validation';
+
+describe( 'Reporting API validation', () => {
+	describe( 'isValidDateRange', () => {
+		it( 'should return TRUE if dateRange is valid only', () => {
+			expect( isValidDateRange( 'last-14-days', '', null ) ).toBeTruthy();
+		} );
+
+		it( 'should return TRUE if startDate and endDate are valid only', () => {
+			expect( isValidDateRange( '', '2020-01-01', '2020-04-05' ) ).toBeTruthy();
```

Comment:
As mentioned, I think changing the function signature to use object args means we don't have to pass an argument to `dateRange`, which is clearer in use. Bonus: it makes it clear which is the start and end date.

Reply:
Updated.